src/nodejs/hooks/seeker_agent_test.go

```diff
@@ -0,0 +1,281 @@
+			oldVcapServices    string
+			oldBpDebug         string
+		)
+		BeforeEach(func() {
+			oldVcapApplication = os.Getenv("VCAP_APPLICATION")
+			oldVcapServices = os.Getenv("VCAP_SERVICES")
+			oldBpDebug = os.Getenv("BP_DEBUG")
+			os.Setenv("SEEKER_APP_ENTRY_POINT", "server.js")
+		})
+		AfterEach(func() {
+			os.Setenv("VCAP_APPLICATION", oldVcapApplication)
+			os.Setenv("VCAP_SERVICES", oldVcapServices)
+			os.Setenv("BP_DEBUG", oldBpDebug)
+		})
+
+		Context("VCAP_SERVICES contains seeker service - as a user provided service", func() {
+			BeforeEach(func() {
+				os.Setenv("VCAP_APPLICATION", `{"name":"pcf app"}`)
+				os.Setenv("VCAP_SERVICES", `{
+																										  "user-provided": [
+																										    {
+																										      "name": "seeker_service_v2",
+																										      "instance_name": "seeker_service_v2",
+																										      "binding_name": null,
+																										      "credentials": {
+																												"seeker_server_url": "http://10.120.9.117:9911"
+																										      },
+																										      "syslog_drain_url": "",
+																										      "volume_mounts": [],
+																										      "label": "user-provided",
+																										      "tags": []
+																										    }
+																										  ]
+																										 }`)
+			})
+			It("Prepends 'require('./seeker/node_modules/@synopsys-sig/seeker);' to the server.js file", func() {
+				entryPointPath := filepath.Join(buildDir, "server.js")
+				Expect(entryPointPath).ToNot(BeAnExistingFile())
+				const mockedCode = "some mock javascript code"
+				Expect(ioutil.WriteFile(entryPointPath, []byte(mockedCode), 0755)).To(Succeed())
+				err = seeker.AfterCompile(stager)
+				Expect(err).To(BeNil())
+				contents, err := ioutil.ReadFile(entryPointPath)
+				Expect(err).To(BeNil())
+				Expect(string(contents)).To(Equal(hooks.SeekerRequire + mockedCode))
+			})
+			It("does not prepend 'require('./seeker/node_modules/@synopsys-sig/seeker);' to the server.js file if it already exist", func() {
+				entryPointPath := filepath.Join(buildDir, "server.js")
+				Expect(entryPointPath).ToNot(BeAnExistingFile())
+				const mockedCode = hooks.SeekerRequire + "some mock javascript code"
+				Expect(ioutil.WriteFile(entryPointPath, []byte(mockedCode), 0755)).To(Succeed())
+				err = seeker.AfterCompile(stager)
+				Expect(err).To(BeNil())
+				contents, err := ioutil.ReadFile(entryPointPath)
+				Expect(err).To(BeNil())
+				Expect(string(contents)).To(Equal(mockedCode))
+			})
+			It("Fails to prepend the require to the server.js file - when the file does not exist", func() {
+				entryPointPath := filepath.Join(buildDir, "server.js")
+				Expect(entryPointPath).ToNot(BeAnExistingFile())
+				err = seeker.AfterCompile(stager)
+				Expect(err).NotTo(BeNil())
+				Expect(err.Error()).To(ContainSubstring("no such file or directory"))
```

Comment:
You can do the error check for this in a single expectation:

```suggestion
				Expect(err).To(MatchError(ContainSubstring("no such file or directory")))
```
Once this change is made, you can remove the `Expect(err).NotTo(BeNil())` assertion above.

Reply:
fixed